addons/mail/static/tests/mail_utils_tests.js

```diff
@@ -35,5 +35,41 @@ QUnit.test('add_link utility function', function (assert) {
     });
 });
 
+QUnit.test('addLink: linkify inside text node (0 occurrence)', function (assert) {
+    assert.expect(1);
+
+    var content = '&amp; &amp;amp; &gt; &lt;';
+    var linkified = utils.parseAndTransform(content, utils.addLink);
+    assert.strictEqual(
+        linkified,
+        content,
+        "linkifying text should not break html entities"
+    );
+});
+
+QUnit.test('inline: inline utility function', function (assert) {
+    assert.expect(1);
+
+    var content = '&amp; &amp;amp; &gt; &lt;';
+    var linkified = utils.parseAndTransform(content, utils.inline);
+    assert.strictEqual(
+        linkified,
+        content,
+        "inlining text should not break html entities"
+    );
+});
+
+QUnit.test('stripHTML: stripHTML utility function', function (assert) {
+    assert.expect(1);
+
+    var content = '&amp; &amp;amp; &gt; &lt;';
+    var linkified = utils.parseAndTransform(content, utils.stripHTML);
+    assert.strictEqual(
+        linkified,
+        content,
+        "stripHTML should not break html entities"
+    );
+});
```

Comment:
`stripHTML` should remove HTML entities as it did before because its goal is to transform HTML into text.

A post-transformation escape might be necessary depending on how this text is then used.

Reply:
for stripHtml I think you are right, the current behavior should be correct (the only usage seems to be in notification and I guess it expects to receive text and not html), I am removing the escaping

Reply:
Notification is actually expecting text or html (depending on if it's a native notification or one displayed in Odoo) but that's another issue to resolve...

Reply:
yes, I added "OPEN tag"t back, it was nice simplification but needed if stripHtml is not fixed